cursive-core/src/views/dialog.rs

```diff
@@ -379,17 +440,58 @@ impl Dialog {
         self.padding.right = padding;
     }
 
-    /// Returns an iterator on this buttons for this dialog.
-    pub fn buttons_mut(&mut self) -> impl Iterator<Item = &mut Button> {
```

Comment:
Between `buttons` / `buttons_mut` and `iter_buttons` / `iter_buttons_mut`, I think I do prefer `buttons`/`buttons_mut` - using simple plurals for iteration is the traditional approach. The only drawback is a potential conflict if we want a `buttons` chainable setters for an iterator of buttons, but I don't think it will happen.

I do agree we might want a non-mutable iterator version.